code/modules/assembly/signaler.dm

```diff
@@ -211,6 +211,11 @@
 		to_chat(user, "<span class='notice'>Analyzing... [src]'s stabilized field is fluctuating along frequency [format_frequency(frequency)], code [code].</span>")
 	..()
 
+///so we can properly override this in-case we need some special stuff passed along with it, like the reagent type with fluescent anomalies. also pass turf because qdel and
+/obj/item/assembly/signaler/anomaly/proc/ReviveAnomaly(turf/T)
+	qdel(src)
+	return new anomaly_type(T)

```

Comment:
Have you tested this locally to work properly?

Reply:
It works properly, but you're right. Running code on qdelled objects feels gross. I'll just use a QDEL_IN with a 0 timer